plugin/storage/grpc/proto/storage.proto

```diff
@@ -134,6 +134,15 @@ message FindTraceIDsResponse {
     ];
 }
 
+// empty; extensible in the future
+message ArchiveSupportedRequest {
+
+}
+
+message ArchiveSupportedResponse {
+    bool supported = 1;
+}
```

Comment:
I don't think this is necessary. The internal grpc plugin client can always pretend that `storage.ArchiveFactory` is implemented, but the grpc plugin server will check if the plugin actually implements archive plugin interface, and if not it can return a not-implemented error.

Reply:
To detect whether grpc plugin support archive storage or not I have to call one of methods: `GetArchiveTrace` or `WriteArchiveSpan`. Writing new span every grpc plugin init is bad. I believe calling `GetArchiveTrace` every grpc plugin init is not ideal way either. Thats why I introduce new `ArchiveSupported` method.

Reply:
I see, you're trying to match the existing factory behavior and return nil from archive factory methods. Fair enough.

I think it makes sense to introduce a new grpc service:

```
message CapabilitiesRequest { }

message CapabilitiesResponse { 
    bool archiveSpanReader = 1;
    bool archiveSpanWriter = 2;
}

service PluginCapabilities {
    rpc Capabilities(CapabilitiesRequest) returns (CapabilitiesResponse);
}
```

This approach is forward-compatible with new storage interfaces we might want to add in the future. The client will query this service at start-up. Internally, the server can implement this method based on the storage implementing the new interface ArchiveStoragePlugin that I showed above.